### ISSUE TITLE
Disabling access logging through Telemetry API does not disable logging for TCP calls

### ISSUE THREAD

**Issue by rolandkool** [istio/istio#37723]
### Bug Description

When disabling the access logging through the Telemetry API, all HTTP calls stop being logged (as expected), but TCP calls are not.
For example, one of out workloads still logs outgoing calls to Google APIs (on port 443), Graphite (port 2003) and Consul (port 8300). All non-HTTP traffic.

The telemetry config to disable the logs looks like:
```
  accessLogging:
  - disabled: true
    providers:
    - name: envoy
```
Access logging has been configured with a custom format and encoding (JSON)

### Version

```prose
Istio ASM 1.12.4
```

**Comment by zirain** [comment]
can you share a your configdump?

**Comment by rolandkool** [comment]
A config_dump for an instance with this issue can be found here: https://gist.github.com/rolandkool/f351483eb7caf0a8065c40ae72ca2a35

**Comment by zirain** [comment]
> A config_dump for an instance with this issue can be found here: https://gist.github.com/rolandkool/f351483eb7caf0a8065c40ae72ca2a35

can you try this with 1.13?

**Comment by rolandkool** [comment]
I cannot reproduce this with 1.13, so I assume it was fixed in that release.

**Comment by zirain** [comment]
> I cannot reproduce this with 1.13, so I assume it was fixed in that release.

yes, I think those fixes did not backport to 1.12

**Comment by Kmoneal** [comment]
Just wanted to check if this is this resolved by https://github.com/istio/istio/pull/36586 and https://github.com/istio/istio/pull/36661?

**Comment by zirain** [comment]
> Just wanted to check if this is this resolved by #36586 and #36661?

I did not reproduce this on v1.13, so I would say `yes`.
